app/flame_chart/profile_model.ts

```diff
@@ -27,43 +27,69 @@ export type ThreadTimeline = {
 };
 
 function eventComparator(a: TraceEvent, b: TraceEvent) {
-  const tid = a.tid !== undefined && b.tid !== undefined ? a.tid - b.tid : 0;
-  if (tid !== 0) return tid;
+  // Group by thread ID.
+  const threadIdDiff = a.tid - b.tid;
+  if (threadIdDiff !== 0) return threadIdDiff;
 
-  const ts = a.ts !== undefined && b.ts !== undefined ? a.ts - b.ts : 0;
-  if (ts !== 0) return ts;
+  // Sort in increasing order of start time.
+  const tsDiff = a.ts - b.ts;
+  if (tsDiff !== 0) return tsDiff;
 
-  const dur = a.dur !== undefined && b.dur !== undefined ? a.dur - b.dur : 0;
-  if (dur !== 0) return dur;
+  // When two events have the same start time, longer events should come first, since
+  // those are considered the parent events of the shorter events and we want to push them
+  // to the stack first.
+  const durationDiff = b.dur - a.dur;
+  return durationDiff;
+}
+
+// Don't show events smaller than 1ms for now.
+const EVENT_VISIBILITY_THRESHOLD_MICROSECONDS = 1 * 1000;
```

Comment:
is there a way to do this filtering based on zoom level? same with below? 

maybe just filter out any events smaller than 1 px?

Reply:
Yeah, we could at least do it based on initial zoom level. Doing it based on the current zoom level (as the user scrolls) is a lot harder though since we'd need to keep track of which elements are currently in view as the user scrolls.

Updated so it figures it out based on the initial zoom level which is a bit more sane than these heuristics. Had to add a "ruler" div which figures out the component's initial dimensions before it actually renders.